pkg/rkeworker/docker.go

```diff
@@ -201,17 +202,23 @@ func changed(ctx context.Context, c *client.Client, expectedProcess v3.Process,
 			expectedEnvs := make(map[string]string, 8)
 			for _, env := range defaultDockerInspect.Config.Env {
 				es := strings.SplitN(env, "=", 2)
-				expectedEnvs[es[0]] = es[1]
+				if len(es) == 2 {
+					expectedEnvs[es[0]] = es[1]
+				}
 			}
 			for _, env := range expectedProcess.Env {
 				es := strings.SplitN(env, "=", 2)
-				expectedEnvs[es[0]] = es[1]
+				if len(es) == 2 {
```

Comment:
this is a valid env var config in a docker inspect:
```
"Env": [
                "foo",
                "foo=bar",
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
            ],
```
When the env var is just a key without a value, it tells docker to inherit the value from the host. Do you want to account for that scenario?

Reply:
@cjellick, yes

Reply:
It will be in the container with the value that that variable had on the host

Reply:
Oh, I got that. You means `foo1` always a field name in `expectedProcess.Env`, but maybe have field value in `actualProcess.Env`.